### ISSUE TITLE
in android 13 Printing.convertHtml() function not working app got stuck at that point and generate nothing!

### ISSUE THREAD

**Issue by bhaumik1111** [DavBfr/dart_pdf#1517]
**Describe the bug**

i have other device samsung c7 pro(android version 8) and in that device this function works perfectly fine but got error in my other device samsung m34.

i share below code i call this lines of code it will print 
htmlToParse:-<html><body><p>Hello!</p></body></html>

then i wait for a 5 min but this function doesn't return any thing in htmlString variable!

i use printing: ^5.11.0 
my flutter sdk 3.10.5

**To Reproduce**
Code snippet to reproduce the behavior:
```dart
htmlToParse="<html><body><p>Hello!</p></body></html>";
            print("htmlToParse:-$htmlToParse");
            htmlString = await Printing.convertHtml(
                format: format,
                html: htmlToParse);
            print("htmlString:-$htmlString");
            
```

how can i solve this issue ??

**Comment by mozomig** [comment]
Same issues

**Comment by bhaumik1111** [comment]
in android 13 device working?

**Comment by mozomig** [comment]
> in android 13 device working?

I have 2 device with 13 Android, Google Pixel and Samsung s22 ultra, and in Samsung that work perfrectly but in Google Pixel method convertHtml is not work it stuck

**Comment by bhaumik1111** [comment]
can you share code?
just that function!

**Comment by andersonmendess** [comment]
same problem here

**Comment by andersonmendess** [comment]
after uninstall the webview update (https://play.google.com/store/apps/details?id=com.google.android.webview) it working again.

**Comment by bhaumik1111** [comment]
Means issue resolve in your app?

**Comment by mozomig** [comment]
> after uninstall the webview update (https://play.google.com/store/apps/details?id=com.google.android.webview) it working again.

ty, it is work for me, bug in webview system it is very funny :D

**Comment by mozomig** [comment]
> Means issue resolve in your app?

it is not application issue, it is issue android webview compinent in devices, all people who get this bug should be remove update android webview component in system.

I think this issue can be close because library make all good

**Comment by bhaumik1111** [comment]
But how??
Can you pls provide step to solve 
I am new to flutter so

**Comment by mozomig** [comment]
> But how?? Can you pls provide step to solve I am new to flutter so
> 
> Actually i also use webview_flutter for webview content in my app

In device where it is not work, need open  https://play.google.com/store/apps/details?id=com.google.android.webview and click button "remove" thats all, dart pdf use PdfDocumentAdapter with system WebView

**Comment by bhaumik1111** [comment]
@mozomig  that's not the solution! for app 
it is not good say to user uninstall the android webview from playstore and it works 

need to find solution from library side!

**Comment by mozomig** [comment]
> @mozomig that's not the solution! for app it is not good say to user uninstall the android webview from playstore and it works
> 
> need to find solution from library side!

Yep, you can write issue for Google, when they fixed webview component and updated it, this issue will be fixed :) it is not issue this library it is issue webview, we don't have solution for fixed in library side it is impossible

**Comment by savs90** [comment]
Does anyone know if there is any ticket that we can track regarding this Android system webview bug? 
I tried with Android system webview beta and dev and they are also not working, so it is not looking good so far.

**Comment by bhaumik1111** [comment]
there is no other option so uninstall Android system webview from device and it will work

**Comment by savs90** [comment]
I did it, but this can't be a serious permanent solution?

**Comment by bhaumik1111** [comment]
i also discuss this point in previous chat.
Now what is the permanent solution that i don't know.
i am also new in flutter so waiting for solution

**Comment by DavBfr** [comment]
I think I'll remove this convertHtml function and point people to another library to do the same thing.

It's not available on all platforms and the result is not consistent. There are bug reports that cannot be fixed because they use the system's WebView to render the HTML, which is vendor-specific.

**Comment by HAFIZkhan-source** [comment]
Any update on this?

**Comment by bhaumik1111** [comment]
Not yet brother!! Developer says android webview side issue and he is right bcoz in ios its working fine!!
So need to wait for android side library side solution!!

**Comment by HAFIZkhan-source** [comment]
@bhaumik1111  But Need to resolve this issue from this package side only. As an app developer, we can't tell app users to remove 
 **android system webview** update from their phone to use this package based feature

**Comment by jelen-HC** [comment]
Any update on a potential fix and/or alternative library?

**Comment by bhaumik1111** [comment]
No brother! @jelen-HC

**Comment by ekuleshov** [comment]
> I think I'll remove this convertHtml function and point people to another library to do the same thing.
> 
> It's not available on all platforms and the result is not consistent. There are bug reports that cannot be fixed because they use the system's WebView to render the HTML, which is vendor-specific.

It will be unfortunate if this functionality is dropped. This is the only option to get a decent PDF that looks the same as the original html in a webview. The suggested alternative is not anywhere close to that and is broken in many ways.

In my tests the current `Printing.convertHtml()` works on iOS, MacOS and Android below 13. Unfortunately it does not work on Android 13 and 14.

However there is a [html_to_pdf](https://github.com/penkzhou/plugin_html_to_pdf) which is also using Android's webview for html to PDF rendering but works on Android 13 and 14.

After a quick comparison I came up with this change below and with this change the `Printing.convertHtml()` now also works on Android 13 and 14 in my tests.

Hopefully you could use this and consider reverting deprecation of this essential feature.

<img width="741" alt="image" src="https://github.com/DavBfr/dart_pdf/assets/440954/65a4ad31-78ff-4547-900b-f3fd9fa31a94">

**Comment by DavBfr** [comment]
@ekuleshov I see. Maybe I can move this code to a new package. This way it's opt-in and obvious that it doesn't work with Windows, Linux, and web.

**Comment by ekuleshov** [comment]
> @ekuleshov I see. Maybe I can move this code to a new package. This way it's opt-in and obvious that it doesn't work with Windows, Linux, and web.

Don'y you already have `Printing.info()` declaring supported capabilities? This doesn't seem different from `canRaster` or `canPrint`.

**Comment by ekuleshov** [comment]
I suppose you can throw an exception saying that feature X is not supported on the current platform. That could be the same of all flags in the `PrintingInfo`.

BTW, for the web support you might be able to use [html2pdf.js](https://ekoopmans.github.io/html2pdf.js/) - it works with jsPDF, which I think you also using.

**Comment by ekuleshov** [comment]
For the web support it seems like [jsPDF](https://github.com/parallax/jsPDF) works well.

It needs these two dependencies:
```
<script src="jspdf.umd.js"></script>
<script src="html2canvas.js"></script>
```

then the following js code produces a very decent PDF document:

```
window.jsPDF = window.jspdf.jsPDF;
var doc = new jsPDF();
doc.html(document.body, {
   callback: function (doc) {
     doc.save();
   },
   width: 135,
   windowWidth: 600
});
```

**Comment by dr953** [comment]
+1 to fix the problem and revert the deprecation...

**Comment by ekuleshov** [comment]
> @ekuleshov I see. Maybe I can move this code to a new package. This way it's opt-in and obvious that it doesn't work with Windows, Linux, and web.

@DavBfr any chance you could incorporate the above changes and revert deprecation of `Printing.convertHtml()`? If it helps I can look at making a PR.

**Comment by DavBfr** [comment]
It would be much better to create a new package. I can create one and move the code over. Then we can improve. Or someone else can do it too.

**Comment by ekuleshov** [comment]
> It would be much better to create a new package. I can create one and move the code over. Then we can improve. Or someone else can do it too.

Sure, if you believe that would be better.
Though it seems like it is going to have big chunk of the same printing boilerplate from Android and iOS.

**Comment by ekuleshov** [comment]
Hi @DavBfr sorry to bug you again about this. Another two months passed and we still don't have a decent solution. None of 3rd party packages provide an acceptable output.

I understand that you want to reduce amount of support, though maybe a one-line change is not too much to consider? Extracting a new module is more work + more support in the long run.

**Comment by dimityrivanov** [comment]
@ekuleshov Hey I have the same issue when using htmlToPdf I've changed the code manually from the library but this function is still disabled in the Info and also the function does not produce any results, did you do anything else then just add this line ?

**Comment by ekuleshov** [comment]
> [@ekuleshov](https://github.com/ekuleshov) ...did you do anything else then just add this line ?

I switched to another plugin for html to pdf stuff, also had to make a custom fork to make it work. Not sure what were the changes now.

**Comment by deakjahn** [comment]
Chiming in. In my case, I create a very simple HTML on the fly, a simple TABLE with UL lists inside. I wouldn't mind a suggestion to a simple package that does the conversion but until then, it would be nice not to see this feature dropped. I'm Android/iOS only with this app, desktop isn't even important for me in this case.
